### PR TITLE
Added job manager class

### PR DESCRIPTION
**Pull request by Onager**
This was inspired by some recent conversations, and #247  - this is manager class for handling job registration. We use a similar scheme in Plaso, and it's been a pretty good model for handling different sorts of subcomponent management (registration/deregistration, enabling/disabling). 

This should go in after #240 and #226 so we don't complicate things. Putting it out now for discussion in advance.

### CODE REVIEW

File: turbinia/jobs/manager.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
+# -*- coding: utf-8 -*-
```
Comment (aarontp) [inline turbinia/jobs/manager.py]: I think we might need the copyright blurb here?

File: turbinia/jobs/manager_test.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+
+  def testGetJobInstance(self):
+    """Tests the GetJobInstance function."""
+    manager.JobsManager.RegisterJob(TestJob1)
+    job = manager.JobsManager.GetJobInstance('testjob1')
+    self.assertIsNotNone(job)
+    self.assertEqual(job.NAME, 'testjob1')
+
+    job = manager.JobsManager.GetJobInstance('testjob1')
```
Comment (aarontp) [inline turbinia/jobs/manager_test.py]: Dupe code block?

File: turbinia/jobs/manager_test.py [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    self.assertIsNotNone(job)
+    self.assertEqual(job.NAME, 'testjob1')
+
+    with self.assertRaises(KeyError):
+      manager.JobsManager.GetJobInstance('bogus')
+    manager.JobsManager.DeregisterJob(TestJob1)
+
+  def testGetJobInstances(self):
+    """Tests getting job objects by name."""
+    job_names = manager.JobsManager.GetJobNames()
```
Comment (aarontp) [inline turbinia/jobs/manager_test.py]: Shouldn't we add a job or two first to really test this?

File: turbinia/jobs/sshd.py [hunk 4]
```diff
@@ -24,21 +24,19 @@
 from turbinia.evidence import ExportedFileArtifact
 from turbinia.evidence import RawDisk
 from turbinia.evidence import ReportText
-from turbinia.jobs import TurbiniaJob
+from turbinia.jobs import interface
+from turbinia.jobs import manager
 
 
-class SSHDExtractionJob(TurbiniaJob):
+class SSHDExtractionJob(interface.TurbiniaJob):
```
Comment (aarontp) [inline turbinia/jobs/sshd.py]: Don't we need to set a NAME for this Job as well?

**Comment by aarontp** [comment]
Did your changes get pushed?  I don't see them in the commit list or the diff.

**Comment by Onager** [comment]
Closing this out, as the git history got screwed up.

**Pull request closed without merging**